nixos/modules/services/misc/p2pool.nix

```diff
@@ -0,0 +1,184 @@
+        default = "your-wallet";
+        description = "Wallet address to mine to. Subaddresses and integrated addresses are not supported!";
+      };
+
+      host = mkOption {
+        type = types.str;
+        default = "";
+        example = literalExpression "127.0.0.1";
+        description = "IP address of your Monero node.";
+      };
+
+      rpcPort = mkOption {
+        type = types.port;
+        default = 18081;
+        example = literalExpression "18081";
+        description = "monerod RPC API port number.";
+      };
+
+      zmqPort = mkOption {
+        type = types.port;
+        default = 18083;
+        example = literalExpression "18081";
+        description = "monerod ZMQ API port number.";
+      };
+
+      stratumServers = mkOption {
+        type = types.listOf serverConfigType;
+        default = [ ];
+        example = literalExpression ''
+          [
+            {
+              address = "127.0.0.1";
+              port = 65535;
+            }
+          ];
+        '';
+        description = "List of IP:port for stratum server to listen on.";
+      };
+
+      p2pServers = mkOption {
+        type = types.listOf serverConfigType;
+        default = [ ];
+        example = literalExpression ''
+          [
+            {
+              address = "127.0.0.1";
+              port = 65535;
+            }
+          ];
+        '';
+        description = "List of IP:port for p2p server to listen on.";
+      };
+
+      p2poolNodes = mkOption {
+        type = types.listOf serverConfigType;
+        default = [ ];
+        example = literalExpression ''
+          [
+            {
+              address = "127.0.0.1";
+              port = 65535;
+            }
+          ];
+        '';
+        description = "List of IP:port for p2pool nodes to connect to.";
+      };
+
+      lightMode = mkOption {
+        type = types.bool;
+        default = false;
+        example = literalExpression "false";
+        description = "Don't allocate RandomX dataset, saves 2GB of RAM.";
+      };
+
+      logLevel = mkOption {
+        type = types.addCheck types.int (x: x >= 0 || x <= 6);
+        default = 6;
+        example = literalExpression "6";
+        description = "Verbosity of the log, integer number between 0 and 6.";
+      };
+
+      noCache = mkOption {
+        type = types.bool;
+        default = false;
+        example = literalExpression "false";
+        description = "Disable p2pool.cache";
+      };
+
+      noColor = mkOption {
+        type = types.bool;
+        default = false;
+        example = literalExpression "false";
+        description = "Disable colors in console output.";
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.services.p2pool = {
+      wantedBy = [ "multi-user.target" ];
+      after = [ "network.target" ];
+      description = "P2Pool Mining Software";
+      serviceConfig = {
+        ExecStart = ''
+          ${pkgs.p2pool}/bin/p2pool \
```

Comment:
Create an extra option for flag list then use that with the rest of the options then you can unwrap that list all at once on the command.

Reply:
I used this service as an example: https://github.com/NixOS/nixpkgs/blob/40bb731ec71bdd7e85ae8c00208d90042e98ce21/nixos/modules/services/networking/monero.nix

Do you think it's worth it creating a new list for this?